### PR TITLE
CI goes vroom again

### PR DESCRIPTION
**Pull request by robertbastian**
See #3070

### CODE REVIEW

File: .github/workflows/artifacts-build.yml [hunk 1]
```diff
@@ -195,36 +194,349 @@ jobs:
         echo "https://storage.googleapis.com/icu4x-pr-artifacts/gha/${{ github.sha }}/wasm-demo/index.html"
         echo "::endgroup::"
 
+  bench-perf:
```
Comment (robertbastian) [inline .github/workflows/artifacts-build.yml:197]: Continuing discussion from #3070:

> Putting the post-merge benchmark jobs into the Artifacts Build workflow doesn't seem clean

- @echeran 

Benchmarking is not primarily about building an artifact, but it does build an artifact (the HTML page) that we want to include in the publish job, so it needs to be in the same workflow as the other artifact-build jobs. Maybe we should rename this to 'Artifacts & Bench'? Alternatively, as you proposed, we call this 'Post merge' and either (a) just live with it that FFI doc and rustdoc run on every commit, or (b) not run them on every commit (I'm not sure how much value these add).

File: .github/workflows/artifacts-build.yml [hunk 2]
```diff
@@ -195,36 +194,349 @@ jobs:
...
+    - name: Run benchmark
+      run: |
+        pushd $PWD && cd ${{ matrix.component }};
+        export REL_OUTPUT_PATH="`dirs +1`/benchmarks/perf/${{ matrix.component }}";
+        mkdir -p $REL_OUTPUT_PATH;
+        export OUTPUT_PATH_CMD="ls -d $REL_OUTPUT_PATH";
+        export OUTPUT_PATH=$(echo $OUTPUT_PATH_CMD | sh);
+        cargo bench --features bench -- --output-format bencher | tee $OUTPUT_PATH/output.txt;
+
+    - name:  Download previous benchmark data
```
Comment (robertbastian) [inline .github/workflows/artifacts-build.yml:250]: I've changed this to use the GCP bucket for a couple of reasons:
* Using the `unmerged-pr-bench-data` branch for PR-level benches is my suspect for why the `datasize` and `locid` benches are reporting errors. I *think* they are using the previous commit on that branch as the baseline data, but that data could be from another PR, or even from the future invocation on the same PR (there can be all kinds of race conditions).
* Using Git branches to store this data is unnecessary, as we only ever need the latest version (it includes the whole history). Having O(10k) commits on these branches makes cloning the repo expensive, and `git log` hard to handle.
* There's no easy way to associate a data commit on the `merged-bench-data` branch with its commit. `main` *might* correspond to `merged-bench-data`, but it takes ~half an hour for that to update, and there can be race conditions if multiple PRs are merged.

File: .github/workflows/artifacts-build.yml [hunk 3]
```diff
@@ -195,36 +194,349 @@ jobs:
...
+      run: |
+        mkdir -p benchmarks/perf
+        gsutil -m cp -rn gs://${{ env.GCP_BUCKET_ID }}/gha/$(git log --pretty=format:'%H' -1 -r HEAD^)/benchmarks/perf/${{ matrix.component }} benchmarks/perf
```
Comment (robertbastian) [inline .github/workflows/artifacts-build.yml:253]: Note: before merging this, we have to upload `merged-bench-data` to GCP under `main`'s current hash.

File: .github/workflows/build-test.yml [hunk 4]
```diff
@@ -47,9 +47,16 @@ env:
 
 jobs:
 
-  # Build job - basic smoke test
-  check-msrv:
+  # ci-job-check & ci-job-features
+  # Running at MSRV (1.61)
+  msrv:
```
Comment (Manishearth) [inline .github/workflows/build-test.yml:52]: nit: Can you provide a `ci-job-msrv` task that just prints out "please run `ci-job-features` or  `ci-job-check` with +<msrv>"

That way we retain the property that `ci-job-foo` is helpful when CI task `foo` fails.

File: components/collections/codepointtrie_builder/Cargo.toml [hunk 5]
```diff
@@ -56,3 +56,7 @@ features = [
 
 [dev-dependencies]
 icu = { path = "../../icu" }
+
+[package.metadata.cargo-all-features]
+# No features, so disable cargo-all-features
+skip_feature_sets = [[]]
```
Comment (Manishearth) [inline components/collections/codepointtrie_builder/Cargo.toml:62]: nit: newline at eof

**Review by Manishearth — DISMISSED** [review]
r=me for main ci changes, i'm not super familiar with the bench stuff

File: .github/workflows/build-test.yml (hunk 4 above)
Comment (robertbastian) [reply .github/workflows/build-test.yml:52]: Added a job that actually does it.

**Review by Manishearth — DISMISSED** [review]

File: Makefile.toml [hunk 6]
```diff
@@ -120,7 +120,7 @@ category = "CI"
 script_runner = "@duckscript"
 script = '''
 exec rustup install 1.61 --profile minimal
-exec cargo +1.61 check-all-features
+exec cargo +1.61 make check-all-features
```
Comment (Manishearth) [inline Makefile.toml:123]: sliightly worried that `cargo +foo make` won't work well all the time due to the toolchain bopping between cargo-make and cargo (we've had problems with this before, and the cargo team has said that cargo-make's use case is somewhat niche here)

File: .github/workflows/artifacts-build.yml (hunk 1 above)
Comment (echeran) [reply .github/workflows/artifacts-build.yml:192]: The current name is fine and makes sense -- dashboards are artifacts, too.

**Review by echeran — DISMISSED** [review]
LGTM just looking at the benchmark stuff. Deferring to Manish's approval for the rest.

File: Makefile.toml (hunk 6 above)
Comment (robertbastian) [reply Makefile.toml:123]: If I make the `check-all-features` job do `cargo --version` this gives me 1.61. Anyway, this is only a problem for the features job, most MSRV violations will be caught by the check job which doesn't recursively call `cargo-make`.

**Review by Manishearth — DISMISSED** [review]

**Comment by Manishearth** [comment]
Force merging due to branch protection rule

**Pull request merged**